### ISSUE TITLE
[doc] Error/warning on non-integer period on wave

### ISSUE THREAD

**Issue by sjgitty** [lowRISC/opentitan#80]
> .../docgen.py spi_device.md > spi_device.html
ERROR: Clock must have an integer period

Seems to work though. Perhaps WARN or perhaps a real error lurking?

**Comment by towoe** [comment]
I had a look into this and I'm not completely sure I get it.
According to wavedrom schema, period can only be a natural number, and the clock indicator should be used only at the beginning of the string.
So my guess is we ignored those "rules" because it does work and the images look good?
And the reason for this was to have the extended cycle with the gap, which resulted in two cycles?
An alternative could be to use the barely documented sub-cycle feature https://github.com/wavedrom/wavedrom/issues/243 but I think it would still require some upstream work in order to remove some artifacts which are the result of not having the clock indicator at the first position and some sub-cycle resetting.

Current:
![2020-05-19-132407](https://user-images.githubusercontent.com/8666134/82320918-58ca7080-99d4-11ea-8089-7869ff6ad78e.png)

Modified:
![2020-05-19-132435](https://user-images.githubusercontent.com/8666134/82320926-5c5df780-99d4-11ea-9bbf-b2086db7ff39.png)

**Comment by sjgitty** [comment]
@towoe yes you have the right way of looking at it. We used the half period because it happened to work, but gave the error in these few cases. I don't know if we care enough tbh to fix the source, but your proposed change looks "good enough". You could even remove the `<` and `>` for CSB to get rid of the unsightly "glitch", though it makes the S slightly misaligned.

**Comment by rswarbrick** [comment]
The OpenTitan project doesn't actually use Wavedrom any more, since b5d5dec88cb806601852ac553ae0ca6d7505cc00. We've still got a few chunks of the word (vendored in from Ibex), but I don't think there's anything left to do here for the project.

**Comment by towoe** [comment]
> The OpenTitan project doesn't actually use Wavedrom any more, since [b5d5dec](https://github.com/lowRISC/opentitan/commit/b5d5dec88cb806601852ac553ae0ca6d7505cc00). We've still got a few chunks of the word (vendored in from Ibex), but I don't think there's anything left to do here for the project.

Awesome to see that also such old issues are being cleaned-up with a meaningful comment. Great project, thanks Rupert!
